### PR TITLE
Front : Display video duration

### PR DESCRIPTION
**Pull request by sandre35**
Warning: this PR needs to be merged first : https://github.com/tournesol-app/tournesol/pull/318
Linked issue : https://github.com/tournesol-app/tournesol/issues/282
This PR tackles steps 3 and 4.

### CODE REVIEW

**Comment by sandre35** [comment]
Screenshot from 2021-11-26 16-12-58
the first video has the duration registered in the db, contrary to the second one

File: frontend/yarn.lock [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 "@apidevtools/json-schema-ref-parser@9.0.9":
   version "9.0.9"
-  resolved "https://registry.yarnpkg.com/@apidevtools/json-schema-ref-parser/-/json-schema-ref-parser-9.0.9.tgz#d720f9256e3609621280584f2b47ae165359268b"
+  resolved "https://registry.npmjs.org/@apidevtools/json-schema-ref-parser/-/json-schema-ref-parser-9.0.9.tgz"
```
Comment (lfaucon) [inline frontend/yarn.lock:7]: You probably should not have these changes. Did you run `npm install` instead of `yarn install` ?

**Review by lfaucon — CHANGES_REQUESTED** [review]
Could we display the duration more similarly as on Youtube? Maybe there is an option with ReactPlayer ? Otherwise, we may write our own VideoPlayer component to be used in VideoCard, VideoSelector & VideoThumb.

Screenshot 2021-11-26 at 21 49 09

**Comment by sandre35** [comment]
@lfaucon there is no parameter that enables us to print the duration of the video on preview with ReactPlayer options.

File: frontend/scripts/openapi.yaml [hunk 2]
```diff
@@ -10567,8 +10567,13 @@ components:
           type: integer
           readOnly: true
           description: Total number of certified contributors who rated the video
+        duration:
+          type: integer
+          readOnly: true
+          description: Duration of the video
```
Comment (amatissart) [inline frontend/scripts/openapi.yaml:10573]: This file does not seem to be up to date with the API. This field `duration` is supposed to be "nullable".

File: frontend/src/features/videos/VideoCard.tsx [hunk 3]
```diff
@@ -221,6 +223,11 @@ function VideoCard({
           {video.uploader && (
             <span className={classes.channel}>{video.uploader}</span>
           )}
+          {!!video.duration && video.duration && (
+            <span className={classes.youtube_complements_p}>
+              {shortEnglishHumanizer(video.duration * 1000)}
+            </span>
+          )}
```
Comment (amatissart) [inline frontend/src/features/videos/VideoCard.tsx:230]: After some experiments, I end up with a potential solution to display the duration in the bottom-right corner of the player, using the [`wrapper` prop](https://github.com/cookpete/react-player#props).

Here is what the wrapper could look like:
```tsx
const PlayerWrapper = ({
  duration,
  children,
}: {
  duration?: string;
  children: React.ReactNode;
}) => {
  const [isDurationVisible, setIsDurationVisible] = useState(true);
  return (
    <Box
      position="relative"
      height="100%"
      onClick={() => setIsDurationVisible(false)}
    >
      {isDurationVisible && duration && (
        <Box
          position="absolute"
          bottom={0}
          right={0}
          bgcolor="rgba(0,0,0,0.5)"
          color="#fff"
          px={1}
          fontSize="0.8em"
          fontWeight="bold"
        >
          {duration}
        </Box>
      )}
      {children}
    </Box>
  );
};
```

**Review by lfaucon — APPROVED** [review]
I changed `convertDurationToClockDuration` to not display hours when the duration is < 1 hour. Thanks for this, it looks great :)

File: frontend/src/features/videos/VideoCard.tsx [hunk 4]
```diff
@@ -125,6 +127,39 @@ const useStyles = makeStyles((theme) => ({
...
+      {isDurationVisible && duration && (
+        <Box
+          position="absolute"
+          bottom={0}
+          right={0}
+          bgcolor="rgba(0,0,0,0.5)"
+          color="#fff"
+          px={1}
+          fontSize="0.8em"
+          fontWeight="bold"
```
Comment (amatissart) [inline frontend/src/features/videos/VideoCard.tsx:153]: A small bug: the duration block will disappear if you click on it. We would probably prefer that pointer events pass through this block.

```suggestion
          fontWeight="bold"
          style={{ pointerEvents: 'none' }}
```

File: frontend/src/features/videos/VideoCard.tsx [hunk 5]
```diff
@@ -65,6 +66,7 @@ const useStyles = makeStyles((theme) => ({
   },
   channel: {
     textDecorationLine: 'underline',
+    marginRight: '12px',
```
Comment (amatissart) [inline frontend/src/features/videos/VideoCard.tsx:69]: No longer relevant in this PR.

**Review by amatissart — APPROVED** [review]

**Pull request merged**